src/libbson/src/jsonsl/jsonsl.c

```diff
@@ -457,13 +457,17 @@ jsonsl_feed(jsonsl_t jsn, const jsonsl_char_t *bytes, size_t nbytes)
                 } else if (state->special_flags & JSONSL_SPECIALf_NULL ||
                            state->special_flags & JSONSL_SPECIALf_NAN) {
                    /* previous char was "n", are we parsing null or nan? */
-                   if (CUR_CHAR != 'u') {
+                   bool not_u = CUR_CHAR != 'u';
+                   bool not_a = tolower (CUR_CHAR) != 'a';
+                   if (not_u) {
                       state->special_flags &= ~JSONSL_SPECIALf_NULL;
                    }
-
-                   if (tolower(CUR_CHAR) != 'a') {
+                   if (not_a) {
                       state->special_flags &= ~JSONSL_SPECIALf_NAN;
                    }
+                   if (not_u && not_a) {
+                      VERIFY_SPECIAL ("null", 4);
+                   }
```

Comment:
Confirming: is `not_u` really supposed to be in this conditional? Given we are verifying the value to be `"null"`, the check being conditioned on `not_u` seems contradictory at first glance.

Reply:
I forgot to put a comment explaining this. The `VERIFY_SPECIAL` here will unconditionally generate an error about an invalid "special" token. This is to halt JSON tokenization if a bare `n` is followed by anything other than a `u` or an `a`. The choice of `"null"` as the string here is just to match what the user more likely meant to type (as NaN is an extension in jsonsl).